internal/liqonet/route-operator/overlayOperator.go

```diff
@@ -104,8 +110,8 @@ func (ovc *OverlayController) Reconcile(ctx context.Context, req ctrl.Request) (
 }
 
 // NewOverlayController returns a new controller ready to be setup and started with the controller manager.
-func NewOverlayController(podName, podIP string, podSelector *metav1.LabelSelector,
-	vxlanDevice overlay.VxlanDevice, cl client.Client) (*OverlayController, error) {
+func NewOverlayController(podName, podIP string, podSelector *metav1.LabelSelector, vxlanDevice overlay.VxlanDevice,
+	nodesLock *sync.RWMutex, vxlanNodes map[string]string, cl client.Client) (*OverlayController, error) {
```

Comment:
Can this mutex be created inside this function instead of in the outside and then passed by reference? I think that there is no reason to share the mutex with other components, and to pass it by reference sounds strange to me.
Or am I missing something?

Reply:
The mutex is used for the `vxlanNodes` map. It is shared by two different go routines. To each go routine we passe the mutex and the map. A better solution would be to create a new `data structure` that holds both the map and the mutex. For the moment i would prefer to keep it simple.

Reply:
ok, thanks